### PR TITLE
Expose metrics of Caffeine caches

### PR DESCRIPTION
**Pull request by merlimat**
### Motivation

Collect and expose metrics related to the Caffeine and Guava caches in brokers. 

This will report the sizes, hit/misses rates and load latency.

### CODE REVIEW

File: pulsar-zookeeper-utils/src/main/java/org/apache/pulsar/stats/CacheMetricsCollector.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class CacheMetricsCollector {
+
+    public static final io.prometheus.client.cache.caffeine.CacheMetricsCollector CAFFEINE = new io.prometheus.client.cache.caffeine.CacheMetricsCollector()
+            .register();
+
+    public static final io.prometheus.client.guava.cache.CacheMetricsCollector GUAVA = new io.prometheus.client.guava.cache.CacheMetricsCollector()
```
Comment (jerrypeng) [inline pulsar-zookeeper-utils/src/main/java/org/apache/pulsar/stats/CacheMetricsCollector.java]: Where is this being used?

Comment (merlimat) [reply pulsar-zookeeper-utils/src/main/java/org/apache/pulsar/stats/CacheMetricsCollector.java]: I was initially planning to instruments also Guava caches, though the remaining ones we have are not very interesting, so dropped the ball. 

I have removed guava now. PTAL.

**Comment by merlimat** [comment]
run java8 tests
run integration tests

**Comment by merlimat** [comment]
run integration tests

**Pull request merged**